common/src/stack/command/stack/commands/set/host/power/imp_ipmi.py

```diff
@@ -0,0 +1,49 @@
+# @copyright@
+# Copyright (c) 2006 - 2019 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+import shlex
+from stack.util import _exec
+from stack.exception import CommandError
+from stack import api
+
+class Implementation(stack.commands.Implementation):
+	def run(self, args):
+		host = args[0]
+		cmd = args[1]
+		ipmi_ip = ''
+		username = ''
+		password = ''
+
+		for interface in api.Call('list.host.interface', args = [host]):
+			if interface['interface'] == 'ipmi':
+				ipmi_ip = interface['ip']
+		if not ipmi_ip:
+			raise CommandError(self, f'{host} missing ipmi interface.')
+
+		try:
+			username = api.Call('list.host.attr', args = [host, 'attr=ipmi_username'])[0].get('value')
```

Comment:
This should be using `self.owner.call` to run the relevant stack commands. I see this repeated, so check throughout.